### PR TITLE
New accessors `roomKeyRequestsEnabled` and `roomKeyForwardingEnabled`

### PR DESCRIPTION
**Pull request by richvdh**
Follow-up to https://github.com/matrix-org/matrix-rust-sdk/pull/2902. Required for https://github.com/vector-im/element-web/issues/26524.

### CODE REVIEW

File: .cargo/config [hunk 1]
```diff
@@ -1,2 +1,9 @@
+[patch.'https://github.com/matrix-org/matrix-rust-sdk']
+matrix-sdk-base = { path = "../matrix-rust-sdk/crates/matrix-sdk-base" }
+matrix-sdk-common = { path = "../matrix-rust-sdk/crates/matrix-sdk-common" }
+matrix-sdk-crypto = { path = "../matrix-rust-sdk/crates/matrix-sdk-crypto" }
+matrix-sdk-indexeddb = { path = "../matrix-rust-sdk/crates/matrix-sdk-indexeddb" }
+matrix-sdk-qrcode = { path = "../matrix-rust-sdk/crates/matrix-sdk-qrcode" }
```
Comment (bnjbvr) [inline .cargo/config:6]: This might not build for everyone (including CI)?

File: src/machine.rs [hunk 2]
```diff
@@ -172,6 +172,38 @@ impl OlmMachine {
...
+    /// have missed in the original share using `m.room_key_request`
+    /// events.
+    #[wasm_bindgen(getter, js_name = "roomKeyRequestsEnabled")]
+    pub fn are_room_key_requests_enabled(&self) -> bool {
+        self.inner.are_room_key_requests_enabled()
+    }
+
+    /// Enable or disable automatic transmission of room key requests.
+    #[wasm_bindgen(setter, js_name = "roomKeyRequestsEnabled")]
+    pub fn toggle_room_key_requests(&self, enabled: bool) {
```
Comment (bnjbvr) [inline src/machine.rs:187]: (here and below for the other function) uber-nit: admittedly I'm being a bit pedantic here, but to me `toggle` carries the meaning of setting the value to `!value`. Would it make sense to replace the prefix with `set_` or `enable_` instead?

File: tests/machine.test.js [hunk 3]
```diff
@@ -201,6 +201,20 @@ describe(OlmMachine.name, () => {
         expect(await machine().displayName).toBeUndefined();
     });
 
+    test("can toggle room key requests", async () => {
+        const m = await machine();
+        expect(m.roomKeyRequestsEnabled).toBeTruthy();
```
Comment (bnjbvr) [inline tests/machine.test.js:206]: Curious, why do we not use true and false in these tests? We're not dealing with null, undefined et al., so a triple-equal comparison would return what we expect.

File: .cargo/config (hunk 1 above)
Comment (richvdh) [reply .cargo/config:6]: oh sorry, that wasn't meant to be here.

**Pull request merged**